src/update-repo-task-status.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  AnalysisStatus,
+  getRepoTask,
+  setVariantAnalysisCanceled,
+  setVariantAnalysisFailed,
+} from "./gh-api-client";
+import {
+  getControllerRepoId,
+  getRepos,
+  getVariantAnalysisId,
+  getWorkflowStatus,
+} from "./inputs";
+
+// All states that indicate the repository has been scanned and cannot
+// change status anymore.
+const COMPLETED_STATES = ["succeeded", "failed", "canceled", "timed_out"];
```

Comment:
Do you think this would be more resilient to change if we turn this around and list all statuses in which the variant analysis is still incomplete (i.e. `pending` and `in_progress`)? If we were to add a status, I would expect it to be a final status (for example another error) rather than a non-final status.

Reply:
Yes! Good point, thanks ✨